### PR TITLE
Build using docker in the makefile

### PR DESCRIPTION
**Pull request by dnephin**
So that the image is only built once

### CODE REVIEW

**Comment by moxiegirl** [comment]
@mikedougherty Hi Mike, we ran into an issue with the build looping.  Daniel suggests this change which does fix it.  It also really changes your `build` target.  Can you give the PR a quick check just to see if you, as the original architect, see any problems with it?

**Comment by SvenDowideit** [comment]
oh wow - @dnephin that's a nice idea.

**Comment by mikedougherty** [comment]
sure, that sounds great actually. i guess i went too far drinking the `compose` koolaid :)

**Comment by thaJeztah** [comment]
nice one, @dnephin !

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 PROJECT_NAME ?= docsdockercom
 DOCKER_COMPOSE := docker-compose -p $(PROJECT_NAME)
+DOCKER_IMAGE := docsdockercom:latest
```
Comment (SvenDowideit) [inline Makefile]: and for the build system, we need this IMAGE name to be set dynamically to something with the GIT-HASH. (as we can be building more than one version at the same time)

Comment (moxiegirl) [inline Makefile]: Per @dnephin "If you don't mind using docker-compose 1.5.0rc1 it supports variables in the  docker-compose.yml so you could use a variable for the image name or tag"  -- is that what you want?

**Comment by moxiegirl** [comment]
LGTM for me. Based on what everyone is saying...LGTM all around.

File: Makefile (hunk 1 above)
Comment (dnephin) [inline Makefile]: i'll open a new PR which uses the environment variable for the tag to support this

**Pull request merged**